theano/tensor/tests/test_basic.py

```
@@ -6747,6 +6747,15 @@ def test_unbroadcast_addbroadcast(self):
 
         assert unbroadcast(addbroadcast(x, 0), 0) is x
         assert addbroadcast(unbroadcast(x, 0), 0) is not x
+        
+        #test for negative indices
+        assert addbroadcast(x, -1) is not x
+        assert addbroadcast(x, -2) is not x
+        assert addbroadcast(x, -3) is x
```

This should raise an error as this is an invalid case.